configs/SPRACINGH7EVO/config.h

```diff
@@ -0,0 +1,284 @@
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define FC_TARGET_MCU           STM32H730
+
+#define BOARD_NAME              SPRACINGH7EVO
+#define MANUFACTURER_ID         SPRO
+
+#define TARGET_BOARD_IDENTIFIER "SP7V"
+#define USBD_PRODUCT_STRING     "SPRacingH7EVO"
+
+#define FC_VMA_ADDRESS          0x90100000
+
+#define EEPROM_SIZE             8192
+
+#define USE_SPRACING_PERSISTENT_RTC_WORKAROUND
+
+// No physical buttons, BIND pad on bottom can be connected though a 10k resistor and switch to V3P3+
+#define USE_BUTTONS
+#define BUTTON_A_PIN            PC14
+#define BUTTON_A_PIN_INVERTED
+#define BUTTON_B_PIN            PC14
+#define BUTTON_B_PIN_INVERTED
+
+// Device has 2x flash chips in parallel using a single CS pin, initial support is for the first flash chip only.
+#define USE_OCTOSPI
+#define USE_OCTOSPI_DEVICE_1
+
+#define USE_SPI
+
+// On mezzanine expansion connector, for external flash for logging, second gyro, sx1280, etc
+#define USE_SPI_DEVICE_1
+#define SPI1_SCK_PIN            PA5
+#define SPI1_SDI_PIN            PB4
+#define SPI1_SDO_PIN            PB5
+#define SPI1_NSS_PIN            PA15
+
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN            PD3
+#define SPI2_SDI_PIN            PB14
+#define SPI2_SDO_PIN            PB15
+#define SPI2_NSS_PIN            PB12
+
+// SX1280 not installed, signals on mezzanine connector
+#define SX1280_BUSY_PIN         PC7
+#define SX1280_DIO1_PIN         PD4
+#define SX1280_DIO2_PIN         NONE
+#define SX1280_DIO3_PIN         NONE
+
+#define USE_FLASH_MEMORY_MAPPED
+#define FLASH_OCTOSPI_INSTANCE  OCTOSPI1
+#define CONFIG_IN_MEMORY_MAPPED_FLASH
+#define USE_FIRMWARE_PARTITION
+
+#define USE_SDCARD_SDIO
+#define SDCARD_DETECT_PIN       PC13
+#define SDCARD_DETECT_INVERTED
+#define SDIO_DEVICE             SDIODEV_1
+#define SDIO_USE_4BIT           1
+#define SDIO_CK_PIN             PC12
+#define SDIO_CMD_PIN            PD2
+#define SDIO_D0_PIN             PC8
+#define SDIO_D1_PIN             PC9
+#define SDIO_D2_PIN             PC10
+#define SDIO_D3_PIN             PC11
+
+// OCTOSPI pins excluded, as bootloader runs firmware in memory mapped mode
+#define TARGET_IO_PORTA         0xffff
+#define TARGET_IO_PORTB         (0xffff & ~(BIT(2)|BIT(6)))
+#define TARGET_IO_PORTC         0xffff
+#define TARGET_IO_PORTD         (0xffff & ~(BIT(11)|BIT(12)|BIT(13)))
+#define TARGET_IO_PORTE         (0xffff & ~(BIT(2)|BIT(7)|BIT(8)|BIT(9)|BIT(10)))
+#define TARGET_IO_PORTF         0xffff
+#define TARGET_IO_PORTG         0xffff
+#define TARGET_IO_PORTH         0xffff
+
+#define USE_I2C
+
+#define I2C1_SCL_PIN            NONE
+#define I2C1_SDA_PIN            NONE
+
+// on GPS connector and mezzanine expansion connector
+// 10k pull-ups installed
+#define USE_I2C_DEVICE_2
+#define I2C2_SCL_PIN            PB10
+#define I2C2_SDA_PIN            PB11
+
+#define I2C3_SCL_PIN            NONE
```

Comment:
NONE is default #ifndef

Reply:
yes, the target definition is being explicit here.  This is how I prefer the SPRacingH7 targets, they're all the same, same goes for the UARTs.